test/integration/main.c

```diff
@@ -26,42 +26,53 @@
 #include <stdbool.h>
 #include "SystemConfig.h"
 #include "Uart.h"
-#include "function_tests/Uart/test_Uart_init.h"
-#include "function_tests/Uart/test_Uart_write.h"
-#include "function_tests/Uart/test_Uart_read.h"
-#include "function_tests/Uart/test_Uart_writeAsync.h"
-#include "function_tests/Uart/test_Uart_readAsync.h"
 #include <rtems.h>
 #include <rtems/confdefs.h>
 
 static Uart uart0;
 
-int
-main()
+static volatile bool dataSent;
+
+static void
+testCallback(bool* result)
 {
-    if(!test_Uart_init(Uart_Id_0, &uart0)) {
-        return -1;
-    }
-    if(!test_Uart_write(&uart0)) {
-        return -1;
-    }
-    if(!test_Uart_writeAsync(&uart0)) {
-        return -1;
-    }
-    if(!test_Uart_read(&uart0)) {
-        return -1;
-    }
-    if(!test_Uart_readAsync(&uart0)) {
-        return -1;
+    *result = true;
+}
+
+static const Uart_TxHandler testHandler = {
+    .callback = (UartTxEndCallback)testCallback,
+    .arg = &dataSent
+};
+
+BYTE_FIFO_CREATE_FILLED(txByteFifoForWriteTest, "Write text (async)\r\n");
+
+bool
+test_Uart_writeAsync(Uart* uart)
```

Comment:
Isn't this a duplicate?

Reply:
This file is unused, I forgot to delete it.